grails-app/domain/org/pih/warehouse/product/Product.groovy

```diff
@@ -711,17 +715,58 @@ class Product implements Comparable, Serializable {
      * @return the display name if it exists or null
      */
     String getDisplayName() {
-        List<Synonym> synonyms = getSynonymsByLocale(SynonymTypeCode.DISPLAY_NAME, LocalizationUtil.currentLocale)
+        return getDisplayName(LocalizationUtil.currentLocale)
+    }
+
+    String getDisplayName(Locale locale) {
+        List<Synonym> synonyms =  getSynonyms(SynonymTypeCode.DISPLAY_NAME, locale)
         return !synonyms.empty ? synonyms.first().name : null
     }
 
+    List<Synonym> getSynonyms(SynonymTypeCode synonymTypeCode) {
+        return getSynonyms(synonymTypeCode, LocalizationUtil.currentLocale)
+    }
+
     /**
      * @return the display name if it exists or the default name
      */
     String getDisplayNameOrDefaultName() {
         return displayName ?: name
     }
 
+    Map getAlias() {
+        return getAlias(LocalizationUtil.currentLocale)
+    }
+
+    Map getAlias(Locale locale) {
+        return [
+                locale          : locale,
+                displayName     : getDisplayName(locale),
+                alternativeNames: getSynonyms(SynonymTypeCode.ALTERNATE_NAME, locale),
+                brandNames      : getSynonyms(SynonymTypeCode.BRAND_NAME, locale),
+        ]
+    }
+
+    // FIXME Come up with a better way to handle this i.e. store Locale objects in memory
+    Map getAliases() {
+        def supportedLocales = ConfigurationHolder.config.openboxes.locale.supportedLocales
+        def locales = supportedLocales.collect { return new Locale(it) }
```

Comment:
I think we could at least add those two lines to `LocalizationUtil`, as I can see the same logic is applied in the method below.

```groovy
static List<Locale> getSupportedLocales() {
   def supportedLocales = ConfigurationHolder.config.openboxes.locale.supportedLocales
   return supportedLocales.collect { return new Locale(it) }
}
```

and do here:
```suggestion
        List<Locale> supportedLocales = LocalizationUtil.supportedLocales
```